dbt/include/dremio/macros/materializations/incremental/incremental.sql

```diff
@@ -14,64 +14,96 @@ limitations under the License.*/
+  -- grab current tables grants config for comparison later on
   {% set grant_config = config.get('grants') %}
+  {{ drop_relation_if_exists(preexisting_intermediate_relation) }}
+  {{ drop_relation_if_exists(preexisting_backup_relation) }}
+
+  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+ 
+  -- `BEGIN` happens here:
+  {{ run_hooks(pre_hooks, inside_transaction=True) }}
 
-  {{ run_hooks(pre_hooks) }}
+  {% set to_drop = [] %}
 
-  {% if old_relation is none %}
-    {% set build_sql = create_table_as(False, target_relation, external_query(sql)) %}
+  {% if existing_relation is none %}
+    {% set build_sql = get_create_table_as_sql(False, target_relation, sql) %}
   {% elif full_refresh_mode %}
-    {% do adapter.drop_relation(old_relation) %}
-    {% set build_sql = create_table_as(False, target_relation, external_query(sql)) %}
+    {% set build_sql = get_create_table_as_sql(False, intermediate_relation, sql) %}
+    {% set need_swap = true %}
   {% else %}
-    {% if tmp_relation is not none %}
-      {{ adapter.drop_relation(tmp_relation) }}
+    {% do run_query(get_create_table_as_sql(True, temp_relation, sql)) %}
+    {% do adapter.expand_target_column_types(
+             from_relation=temp_relation,
+             to_relation=target_relation) %}
+    -- Process schema changes. Returns dict of changes if successful. Use source columns for upserting/merging
+    {% set dest_columns = process_schema_changes(on_schema_change, temp_relation, existing_relation) %}
+    {% if not dest_columns %}
+      {% set dest_columns = adapter.get_columns_in_relation(existing_relation) %}
     {% endif %}
-    {{ run_query(create_table_as(True, target_tmp_relation, external_query(sql))) }}
-    {{ process_schema_changes(on_schema_change, target_tmp_relation, old_relation) }}
-    {% set build_sql = dbt_dremio_get_incremental_sql(strategy, target_tmp_relation, target_relation, unique_key) %}
+
+    -- Get the incremental_strategy, the macro to use for the strategy, and build the sql
+    {%- set incremental_strategy = config.get('incremental_strategy', validator=validation.any[basestring]) or 'append' -%}
+    {%- set raw_file_format = config.get('format', validator=validation.any[basestring]) or 'iceberg' -%}
+    {%- set file_format = dbt_dremio_validate_get_file_format(raw_file_format) -%}
+    {%- set strategy = dbt_dremio_validate_get_incremental_strategy(incremental_strategy, file_format) -%}
+    {%- set raw_on_schema_change = config.get('on_schema_change', validator=validation.any[basestring]) or 'ignore' -%}
+    {% set build_sql = dbt_dremio_get_incremental_sql(strategy, intermediate_relation, target_relation, unique_key) %}
+    
   {% endif %}
 
   {%- call statement('main') -%}
     {{ build_sql }}
   {%- endcall -%}
 
-  {% if not(old_relation is none or full_refresh_mode) %}
-    {{ adapter.drop_relation(target_tmp_relation) }}
+  {% if need_swap %}

```

Comment:
I noticed rename_relation is being used in view and table macros as well. Is this something we should be doing as well?

Reply:
This might be a separate ticket, but it'll keep our macros more up to date.

Reply:
rename_relation is used here for full_refresh_mode to fully replace the relation. I don't see a real need to use it in other macros unless its functionality is needed.